src/wallet/wallet.cpp

```diff
@@ -3728,25 +3725,29 @@ bool CReserveKey::GetReservedKey(CPubKey& pubkey, bool internal)
         fInternal = keypool.fInternal;
     }
     assert(vchPubKey.IsValid());
-    pubkey = vchPubKey;
+    pwallet->LearnRelatedScripts(vchPubKey, type);
+    address = GetDestinationForKey(vchPubKey, type);
+    dest = address;
     return true;
 }
 
-void CReserveKey::KeepKey()
+void ReserveDestination::KeepAddress()
```

Comment:
We're using "destination" in `GetReservedDestination` but "address" here. We should pick one?

Reply:
Whoops, fixed